app/code/core/Mage/SalesRule/Model/Rule.php

```diff
@@ -367,7 +367,6 @@ public function getCoupons()
     {
         if ($this->_coupons === null) {
             $collection = Mage::getResourceModel('salesrule/coupon_collection');
-            /** @var Mage_SalesRule_Model_Resource_Coupon_Collection */
```

Comment:
Can this be moved one line above?

Reply:
This comment is not needed. Both phpstan and phpstorm knows type of this variable.

Reply:
This comment is not needed. Both phpstan and phpstorm knows type of this variable.